export as JSON process graph (#50)
Open-EO/openeo.org: documentation/1.0/cookbook/README.md

</template>
</CodeSwitcher>

> ### Output as JSON Process Graph

Reviewer: ```suggestion
### Output as JSON Process
```

We don't use "process graph" a lot in external communication, just call it a (user-defined) process here and in the other parts of the documentation.

In some cases we want to export the JSON representation of a process we created. If you followed one of the examples above, then your end node is called `res`. If not, replace that with whatever you called your process end node.

<CodeSwitcher>
<template v-slot:py>

```python

> graph = res.to_json()

Reviewer: ```suggestion
process = res.to_json()
```

# if needed, write graph to file, e.g.:

> with open("./process_graph.json", "w") as tfile:

Reviewer: ```suggestion
with open("./process.json", "w") as tfile:
```

>     tfile.write(graph)

Reviewer: ```suggestion
    tfile.write(process)
```

Reviewer: Please also change the terminology in the other examples.

```

</template>
<template v-slot:r>

```r

> graph <- as(res, "Graph")

Reviewer: The proposed solution here looks different: https://github.com/Open-EO/openeo-r-client/issues/54
Not sure though what works better.

Author: Thanks! "as Graph" is indeed wrong here. Changed it.

# if needed, write graph to file, e.g.:
cat(graphToJSON(graph),file = "./process_graph.json")
```

</template>
<template v-slot:js>

```js
// set last node as result = true so it is recognized as the result node
res.result = true

> graph = JSON.stringify(builder.toJSON(), null, 2)

Reviewer: ```suggestion
process = JSON.stringify(builder.toJSON())
```

// if needed, write graph to file, e.g.:
var fs = require('fs');

> fs.writeFile ("./process_graph.json", graph, function(err) {if (err) throw err;});

Reviewer: For the sake of simplicity:

```suggestion
fs.writeFileSync("./process.json", process);
```

</template>
</CodeSwitcher>

### Output: Process as JSON

In some cases we want to export the JSON representation of a process we created. If you followed one of the examples above, then your end node is called `res`. If not, replace that with whatever you called your process end node.

<CodeSwitcher>
<template v-slot:py>

```python
process = res.to_json()

# if needed, write JSON to file, e.g.:
with open("./process.json", "w") as tfile:
    tfile.write(process)
```

</template>
<template v-slot:r>

```r
process <- create_user_process(res, id = "-Title-", submit = FALSE)

# convert list to JSON
process <- jsonlite::toJSON(process, force = TRUE)

# if needed, write JSON to file, e.g.:
cat(process, file = "./process.json")
```

</template>
<template v-slot:js>

```js
// set last node as result = true so it is recognized as the result node
res.result = true

process = JSON.stringify(builder.toJSON())

// if needed, write JSON to file, e.g.:
var fs = require('fs');
fs.writeFileSync("./process.json", process);
```

> **Note:** The above description only works with NodeJS. To do the same in a browser, methodology must be adjusted.

Reviewer: ```suggestion
**Note:** The code to store a file only works with NodeJS. Browsers can't write files to disk.
```

<template v-slot:js>

```js
// set last node as result = true so it is recognized as the result node
res.result = true

graph = JSON.stringify(builder.toJSON(), null, 2)

> // if needed, write graph to file, e.g.:

Reviewer: This only works in NodeJS; not in a Browser!

Author: Should I add a note or methods for browser?

Reviewer: Yes, please add a note :-)
